app/containers/SongDetailContainer/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ * songDetailContainer Container
+ *
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { injectSaga } from 'redux-injectors';
+import { useParams } from 'react-router-dom';
+import { get, isEmpty } from 'lodash';
+import styled from 'styled-components';
+import T from '@components/T';
+import saga from './saga';
+import { selectItuneData } from '../SongContainer/selectors.js';
+import { selectTrackId, selectSingleItune } from './selectors.js';
+import { songDetailContainerCreators } from './reducer';
+
+const DetailsCard = styled.div`
+  border: 1px solid black;
+  border-radius: 10px;
+  padding: 1rem;
+  width: 70%;
+  height: 15rem;
+  margin: auto;
+  margin-top: 3rem;
+
+  display: flex;
+  gap: 1rem;
+`;
+
+export function songDetailContainer({ ituneData, singleItune, dispatchGetSingleItune }) {
+  const { trackId } = useParams();
+  const songs = get(ituneData, 'results', null);
+  let data = songs?.find((item) => item.trackId === Number(trackId));
+  const songDetails = data ?? (!isEmpty(singleItune) && singleItune.results[0]);
+  const { artistName, trackName, artworkUrl100: imgUrl } = songDetails;
+
+  let dataIsPresent =
+    data || (!isEmpty(singleItune) && singleItune.results.find((item) => item.trackId === Number(trackId)));
```

Comment:
What data will singleItune have? Can it have data for multiple songs? If yes then we should change the name

Reply:
singeItune will have data for a single/individual song